### DISCUSSION TITLE
Cart not found when retrieving order from order ID

### DISCUSSION THREAD

**Discussion by ItzCrazyKns** [swellstores/community#412] [Help] [answered]
Hey,

I am using `swell-js` library to retrieve an order but I have encountered a couple of errors. First of all, in the `swell.cart` namespace, the typings are not complete, it is missing a function called `getOrder`.  Secondly, it returns `Cart not found` error even when the order ID is correct. I have verified that all of my keys and all are correct and works completely fine with other functions

Code:

```tsx
import { useRouter } from "next/router"
import { NextPageContext } from "next";
import swell from '@/lib/swell/index';

const Orders = ({ orderData }: { orderData: swell.Order }) => {
    const router = useRouter()
    const slug = router.query.slug

    return (
        <div>{slug}</div>
    )
}

export const getServerSideProps = async (context: NextPageContext) => {
    const data = await swell.cart.getOrder(context.query.slug as string)
    console.log(data)
    return {
        props: {
            orderDate: data
        }
    }
}

export default Orders
```

**Comment by sags95** [comment, marked as answer]
Hi @ItzCrazyKns are you passing in the order `id` or the `checkout_id` of the cart? The `getOrder()` method requires the `checkout_id`.

**Comment by ItzCrazyKns** [comment]
Oh thank you so much, appreciate that
